src/org/scratchOrgSettingsGenerator.ts

```diff
@@ -229,8 +257,8 @@ export default class SettingsGenerator {
     }
   }
 
-  private createObjectFileContent(
-    name: string,
+  private createRecordTypeAndBusinessProcessFileContent(
```

Comment:
this should be a function,
return assertable json for testing, 
have a typed return value,
not do xml conversion,
and have tests that assert the various outcomes are correctly put in the json

everything that made the package.xml writing and testing awesome applies to the object files, too